pkg/block/fetcher.go

```diff
@@ -533,6 +534,23 @@ func contains(s1 []ulid.ULID, s2 []ulid.ULID) bool {
 	return true
 }
 
+type ReplicaLabelsFilter struct {
```

Comment:
```suggestion
type ReplicaLabelConstructingFakeMetaFilter struct {
```
;p

Reply:
Actually as @kakkoyun found it's not a filter, so maybe different name

Reply:
"Modifier"?

Reply:
When that I've checked it again I realize all other `MetaFetcherFilter`s do similar modifications. To be consistent we can keep the name.